src/csharp/Microsoft.Spark/Sql/Types/ComplexTypes.cs

```diff
@@ -71,7 +73,16 @@ private DataType FromJson(JObject json)
 
         internal override bool NeedConversion() => true;
 
-        internal override object FromInternal(object obj) => throw new NotImplementedException();
+        internal override object FromInternal(object obj)
+        {
+            switch (obj)
+            {
+                case ArrayList objArrayList:
```

Comment:
@elvaliuliuliu - i've put a simple fix in to be able to run collect() on a DataFrame that has a row with an ArrayList - if there is a better way to do this or if something else is more appropriate please let me know

Reply:
@GoEddie To keep the PR small (it's already big :)), can we fix this in a separate PR? I think @elvaliuliuliu may have already started on this work.

Reply:
Yes, I agree with @imback82. Let's handle this threw exception for now and add a comment. I will update it once the `ArrayType` is supported.

Reply:
Sure will take it back out.